Automattic/wp-calypso
# Lift up A4A's layout-related components

Related to https://github.com/Automattic/wp-calypso/pull/97384

## Proposed Changes

This PR lifts up reusable layout-related components from A4A. The changes include:

- Relocate components to `client/layout/multi-sites-dashboard/`. So, the primary changes can be found in this file: https://github.com/Automattic/wp-calypso/pull/97542/files#diff-dc091553ce33fb1199069874c32cc8b32c05e04609e5777f41035675fe170baf.
- Rename all `a4a-*` CSS classes.

---
- [ ] I'll follow up on exclude calypso/a8c-for-agencies/* dependency from `client/layout/multi-sites-dashboard` in https://github.com/Automattic/dotcom-forge/issues/10138.

## Why are these changes being made?


Please refer to the Project Thread pbxlJb-6J5-p2 and the proposal post pfYzsZ-Y4-p2 for context. CC: @Automattic/a4a-genesis 

## Testing Instructions



Calypso Live
* Go to /sites
* Observe no visual regression on mobile and desktop

A4A Live
* Go to /overview
* Click all the top-level menu items 
* Observe no visual regression on mobile and desktop


## Pre-merge Checklist
